From: Sven Eckelmann <sven@narfation.org>
Date: Mon, 21 Nov 2022 19:55:47 +0100
Subject: ethtool: Disable use of flex-array for ethtool_link_settings

The struct ethtool_link_ksettings includes struct ethtool_link_settings -
but not at the end. So sparse creates as warning:

  ./include/linux/ethtool.h:148:38: warning: nested flexible array

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/include/uapi/linux/ethtool.h b/include/uapi/linux/ethtool.h
index 58e587ba04507926ac508ea5940bb8ccb2f051f7..55f8b8c1ce3d9f942bda42ba1806ffbd8b33b6ec 100644
--- a/include/uapi/linux/ethtool.h
+++ b/include/uapi/linux/ethtool.h
@@ -2152,7 +2152,7 @@ struct ethtool_link_settings {
 	__u8	master_slave_state;
 	__u8	rate_matching;
 	__u32	reserved[7];
-	__u32	link_mode_masks[];
+	__u32	link_mode_masks[0];
 	/* layout of link_mode_masks fields:
 	 * __u32 map_supported[link_mode_masks_nwords];
 	 * __u32 map_advertising[link_mode_masks_nwords];
